### PR TITLE
make the first user be an admin, even if seeded

### PR DESCRIPTION
**Pull request by yizhang-zen**
@irwaters @grosser

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -93,6 +93,11 @@ def self.create_or_update_from_hash(hash)
     user
   end
 
+  # first user will be the super admin
+  private_class_method def self.initial_user?
+    !User.exists? || User.limit(2).pluck(:email) == ['seed@example.com']
```
Comment (grosser) [inline app/models/user.rb]: does to db queries atm, but we can reduce that to 1

**Comment by grosser** [comment]
rubocop failed, should be easy to fix

**Comment by grosser** [comment]
the 2 failed tests look like random failures ...

**Comment by irwaters** [comment]
looks like rubocop fix is required, so 👍 on green

File: app/models/user.rb [hunk 2]
```diff
@@ -74,12 +74,12 @@ def self.create_or_update_from_hash(hash)
     user = User.where(external_id: hash[:external_id].to_s).first || User.new
 
     # attributes are always a string hash
-    attributes = user.attributes.merge(hash.stringify_keys) do |key, old, new|
-      if key == 'role_id'
-        if !User.exists? # first user will be the super admin
-          Role::SUPER_ADMIN.id
+    attributes = user.attributes.merge(hash.stringify_keys) do |attribute, old, new|
+      if attribute == 'role_id'
+        if !User.where.not(email: 'seed@example.com').exists?
```
Comment (grosser) [inline app/models/user.rb]: FYI @yizhang-zen updated this to be even more efficient ... only a single query and not actually reading any data from the db

**Pull request merged**